api/envoy/extensions/filters/http/oauth2/v3/oauth.proto

```diff
@@ -123,6 +125,9 @@ message OAuth2Config {
   // Defines how ``client_id`` and ``client_secret`` are sent in OAuth client to OAuth server requests.
   // RFC https://datatracker.ietf.org/doc/html/rfc6749#section-2.3.1
   AuthType auth_type = 11 [(validate.rules).enum = {defined_only: true}];
+
+  // Set to true to enable automatic access token refresh using the associated refresh token, provided that the OAuth
+  google.protobuf.BoolValue use_refresh_token = 12;
```

Comment:
nit: add the end `.` after this sentence. This will be used to generate docs, so the formatter is also important.